Zend/Optimizer/zend_dump.c

```diff
@@ -66,7 +69,8 @@ void zend_dump_const(const zval *zv)
 			fprintf(stderr, " float(%g)", Z_DVAL_P(zv));
 			break;
 		case IS_STRING:
-			fprintf(stderr, " string(\"%s\")", Z_STRVAL_P(zv));
+			escaped_string = php_addcslashes(zv->value.str, "\"\\", 2);
+			fprintf(stderr, " string(\"%s\")", escaped_string->val);
```

Comment:
You shouldn't access `->val` directly, please use `ZSTR_VAL`.

Reply:
Done